homeassistant/components/html5/__init__.py

```diff
@@ -1 +1,48 @@
 """The html5 component."""
+import logging
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PLATFORM, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+from .issues import create_issue
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM = Platform.NOTIFY
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the HTML5 push notification component."""
+    existing_config_entry = hass.config_entries.async_entries(DOMAIN)
+    # Iterate all entries for notify to only get HTML5
+    if Platform.NOTIFY in config:
+        for entry in config[Platform.NOTIFY]:
+            if entry[CONF_PLATFORM] == DOMAIN:
+                # the configuration has already been imported
+                # but the YAML configuration is still present
+                if existing_config_entry:
+                    create_issue(hass, True)
+                    return True
+                hass.async_create_task(
+                    hass.config_entries.flow.async_init(
+                        DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
+                    )
+                )
+                return True
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up HTML5 from a config entry."""
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass, Platform.NOTIFY, DOMAIN, dict(entry.data), {}
```

Comment:
I looked at a number of other notify platforms and many of them pass the config as the config parameter as opposed to the discovery parameter (some pass both) but I couldn't find any passing empty dict as the config. Your changes below to `get_service` switch from the config to the discovery info.  Are you sure this is correct? (I'm not sure myself but `async_load_platform` asserts that the config is not none and AFAICT this is config as opposed to something that has been discovered)

Reply:
I'm not terribly familiar with the architecture of HA, but I have seen from pulling this up in the debugger that passing in the data through the config entry doesn't seem to work. I got this idea from the discord component (https://github.com/home-assistant/core/blob/b9899a441c035bf28aec1e30a5167db31adac580/homeassistant/components/discord/notify.py#L42) and it seems to work correctly.

Reply:
Okay I debugged it again and remember now. The config is only presented when configured using YAML. Discovery info is presented during config entry setup. I changed it to also pass the hass config to the load_platform call so that the notify component is loaded correctly. This seems to be what other components are doing as well.